bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/TestPrecompute.cs

```diff
@@ -39,17 +41,49 @@ public void Test_Precompute_Status_Running()
         public void Test_Precompute_Status_Stopped()
         {
             Precompute precompute = new Precompute();
+            var keypair = ElGamalKeyPair.FromSecret(Constants.TWO_MOD_Q);
 
             CancellationTokenSource cts = new CancellationTokenSource();
 
-            precompute.StartPrecomputeAsync(1, cts.Token);
+            precompute.StartPrecomputeAsync(keypair.PublicKey, cts.Token);
+            var statusRunning = precompute.GetStatus();
             precompute.StopPrecompute();
             Thread.Sleep(1000);
 
+            int count = -1;
+            int queue_size = -1;
+            precompute.GetProgress(out count, out queue_size);
             var status = precompute.GetStatus();
 
+            Assert.AreEqual(5000, queue_size);
+
+            Assert.AreEqual(PrecomputeState.Running, statusRunning.CurrentState);
             Assert.AreEqual(PrecomputeState.UserStopped, status.CurrentState);
         }
 
+        [Test]
+        public void Test_Precompute_Status_Complete()
+        {
+            Precompute precompute = new Precompute();
+            var keypair = ElGamalKeyPair.FromSecret(Constants.TWO_MOD_Q);
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            precompute.StartPrecomputeAsync(keypair.PublicKey, cts.Token, 10);
+            Thread.Sleep(2000);

```

Comment:
Any way to eliminate the thread sleeps or at least decrease the numbers substantially?  I like unit tests to complete in well under a second so I can run them constantly in the background.  Maybe these tests could look for a value to change to indicate they're done?  That would make them more deterministic too.  If not this PR, maybe we could consider for future enhancement.

Reply:
The reason for the sleep in the test is to let the calculations get started.  This test actually is to let to calculations naturally complete before we continue.  That is why the buffer size is being set to 10 and a short delay is being used.

Reply:
Removed the Sleep.